tests/image_tests/test_regression.py

```diff
@@ -1,93 +1,85 @@
 import os
 import yaml
-import shutil
 import subprocess
 import pytest
 import fv3config
+import hashlib
 
 
 TEST_DIR = os.path.dirname(os.path.realpath(__file__))
-REFERENCE_DIR = os.path.join(TEST_DIR, "reference")
-OUTPUT_DIR = os.path.join(TEST_DIR, "output")
 CONFIG_DIR = os.path.join(TEST_DIR, "config")
-SUBMIT_JOB_FILENAME = os.path.join(TEST_DIR, "run_files/submit_job.sh")
-STDOUT_FILENAME = "stdout.log"
-STDERR_FILENAME = "stderr.log"
-MODEL_IMAGE = "us.gcr.io/vcm-ml/fv3gfs-wrapper:gnu7-mpich314-nocuda"
-MD5SUM_FILENAME = "md5.txt"
-
-USE_LOCAL_ARCHIVE = True
-
 config_filenames = os.listdir(CONFIG_DIR)
 
 
-@pytest.fixture
-def model_image():
-    return MODEL_IMAGE
-
-
-@pytest.fixture
-def reference_dir(request):
-    refdir = request.config.getoption("--refdir")
-    if refdir is None:
-        refdir = os.path.join(TEST_DIR, "reference/circleci")
-    return refdir
-
-
 @pytest.fixture(params=config_filenames)
 def config(request):
     config_filename = os.path.join(CONFIG_DIR, request.param)
     with open(config_filename, "r") as config_file:
         return yaml.safe_load(config_file)
 
 
-def test_regression(config, model_image, reference_dir):
-    run_name = config["experiment_name"]
-    run_reference_dir = os.path.join(reference_dir, run_name)
-    run_dir = os.path.join(OUTPUT_DIR, run_name)
-    if os.path.isdir(run_dir):
-        shutil.rmtree(run_dir)
-    os.makedirs(run_dir)
-    run_model(config, run_dir, model_image)
-    md5sum_filename = os.path.join(run_reference_dir, MD5SUM_FILENAME)
-    if not os.path.isfile(md5sum_filename):
-        assert False, (
-            f"reference md5sum does not exist, "
-            f"you can create it with `bash set_reference.sh {reference_dir}`"
-            " after running the model and generating the output directories."
-        )
-    else:
-        check_md5sum(run_dir, md5sum_filename)
-    shutil.rmtree(run_dir)
-
-
-def run_model(config, run_dir, model_image):
-    fv3config.write_run_directory(config, run_dir)
-    subprocess.check_call(
-        [
-            "docker",
-            "run",
-            "-v",
-            f"{run_dir}:/rundir",
-            "--workdir",
-            "/rundir",
-            MODEL_IMAGE,
-            "mpirun",
-            "-n",
-            "6",
-        ]
-        + ["python3", "-m", "mpi4py", "-m", "fv3gfs.wrapper.run"]
-    )
+def md5_from_dir(dir_):
+    md5s = {}
+    for root, dirs, files in os.walk(str(dir_)):
+        for file in files:
+            with open(os.path.join(root, file), "rb") as f:
+                md5 = hashlib.md5()
+                while True:
+                    buf = f.read(2048)
+                    if not buf:
+                        break
+                    md5.update(buf)
+
+            md5s[file] = md5.hexdigest()
```

Comment:
Is `file` unique even when you have the same filename in multiple subfolders? If not please prepend `root` to the key here, otherwise looks good.

Reply:
Sure. I was aware of this, but was just a bit lazy.